src/nvim/tui/tui.c

```diff
@@ -641,12 +645,21 @@ static void update_size(UI *ui)
 {
   TUIData *data = ui->data;
   int width = 0, height = 0;
-  // 1 - try from a system call(ioctl/TIOCGWINSZ on unix)
+
+  // 1 - if starting and the 'columns' and 'lines' options are not
+  // the defaults, use their values
+  if (starting && (Columns != 80 || Rows != 24)) {
```

Comment:
The defaults of 80 and 24 are again used on lines 679-680, so we can at least define them at the top of this function.